package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -83,163 +128,201 @@ def get_or_create_modular_pipeline(
-        Example:
-            >>> modular_pipelines = ModularPipelinesRepository()
-            >>> model_output_node = GraphNode.create_data_node(
-            ...     "data_science.model_output", layer=None, tags=set(), dataset=None
-            ... )
-            >>> modular_pipelines.extract_from_node(model_output_node)
-            'data_science'
-            >>> assert modular_pipelines.has_modular_pipeline("data_science")
-        """
+                modular_pipeline.tags.update(kedro_node.tags)
 
-        # There is no need to extract modular pipeline from parameters
-        # because all valid modular pipelines are encoded in either a TaskNode or DataNode.
-        if isinstance(node, ParametersNode):
-            return None
+                # add input/output datasets of a kedro node as a child
+                # to the modular pipeline
+                self._add_datasets_as_children(
+                    modular_pipeline, kedro_node, modular_pipeline_inputs_outputs
+                )
 
-        modular_pipeline_id = node.namespace
-        if not modular_pipeline_id:
-            return None
+                # update node_modular_pipeline lookup
+                if kedro_node_id not in self.node_mod_pipeline_map:
+                    self.node_mod_pipeline_map[kedro_node_id] = set()
+                self.node_mod_pipeline_map[kedro_node_id].add(modular_pipeline_id)
 
-        modular_pipeline = self.get_or_create_modular_pipeline(modular_pipeline_id)
+        # The line `parent_modular_pipeline_id = ('.'.join(modular_pipeline_id.split('.')[:-1])
+        # if '.' in modular_pipeline_id else None)` is extracting the parent modular pipeline ID
+        # from the given modular pipeline ID.
+        parent_modular_pipeline_id = (
+            ".".join(modular_pipeline_id.split(".")[:-1])
+            if "." in modular_pipeline_id
+            else None
+        )
 
-        # Add the node's registered pipelines to the modular pipeline's registered pipelines.
-        # Basically this means if the node belongs to the "__default__" pipeline, for example,
-        # so does the modular pipeline.
-        modular_pipeline.pipelines.update(node.pipelines)
+        if parent_modular_pipeline_id:
+            parent_modular_pipeline = self.get_or_create_modular_pipeline(
+                parent_modular_pipeline_id
+            )
+            parent_modular_pipeline.pipelines.update(modular_pipeline.pipelines)
+            parent_modular_pipeline.tags.update(modular_pipeline.tags)
 
-        modular_pipeline.tags.update(node.tags)
+            # add current modular pipeline and input/output datasets
+            # of a modular pipeline as a child to the parent modular
+            # pipeline based on the rules
+            self._add_children_to_parent_pipeline(
+                parent_modular_pipeline,
+                modular_pipeline_id,
+                modular_pipeline_inputs_outputs,

```

Comment:
This is very unintuitive - the method is called `_add_children`, but in it we have the totally unrelated code for adding the modular pipeline itself to it's parent's children. Could this be factored out as a separate method which is called right after `_add_children` instead?